src/core/util/options.js

```diff
@@ -248,6 +249,30 @@ function normalizeProps (options: Object) {
 }
 
 /**
+ * Normalize all injections into Object-based format
+ */
+function normalizeInject (options: Object) {
+  const inject = options.inject
+  if (!inject) return
```

Comment:
The normalization is only needed if `inject` is an Array, so we can just return if it is not an Array.